### ISSUE TITLE
Supporting event listeners for switching stages by activityID 

### ISSUE THREAD

**Issue by Hafflgav** [wkspower/wks-platform#136]
As a user I want to be able to switch the stage at any given point in the process. 
Currently this is only possible after User-Tasks have been completed. 

To gain more flexibility it would be great to support any Activity- and EventId.

### Example: 
I want to move to the next stage in my case as soon as the process reaches an intermediate event.  The exemplary process can be seen below. 
<img width="1060" alt="image" src="https://github.com/wkspower/wks-platform/assets/30292227/21517d8c-8f88-45dc-9bd1-0b6db704799a">

**Comment by francav** [comment]
I've been exploring ways to effectively track non-human task event types, especially given that their lifecycles aren't currently traceable via the WKS UI and API. One approach that caught my attention was possibly modifying the engine by attaching activity/event listeners, but I have reservations about this due to potential complications.

On a related note, I observed an interesting method in the employee-onboarding demo process. This approach involves using a message-started event-subprocess and dispatching messages from activity listeners to signal stage transitions. Here's a brief outline:

1. Set up activities to trigger messages that would indicate a stage transition.
<img width="1385" alt="Screenshot 2023-08-29 at 20 11 10" src="https://github.com/wkspower/wks-platform/assets/85225281/5908fe6b-b5c6-4536-9761-a9bea0dfac8d">

2. Harness these messages with an event-subprocess and subsequently leverage the standard case stage update external task.
<img width="1380" alt="Screenshot 2023-08-29 at 20 14 39" src="https://github.com/wkspower/wks-platform/assets/85225281/3fa71a87-2158-4189-be9f-911a6ac6f1be">

I'd appreciate any thoughts or alternative suggestions on this. Do you think there might be other effective ways to tackle this challenge?

**Comment by Hafflgav** [comment]
You are right @francav! The easiest possibility to achieve something like this would be by modifying the Camunda Platform 7 engine and adding a listener or some type of exporter for the events. 

For sure this makes this increases the complexity and maintenance of the engine. But in case you are thinking about customising it anyways (e.g. with certain plugins) it might be worth a thought. 

And yes, by using the approach with the event based subprocess this is not needed 👍 
To me this is a valid work around. :)

**Comment by francav** [comment]
Let’s keep this one open for discussion
